### PR TITLE
changed models in database package to correclate User table to Game t…

### PR DESCRIPTION
**Pull request by drew138**
…able via a foreign key

### CODE REVIEW

File: api/status/response.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package status
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+var validationError string = "Validation Error"
+var serverError string = "Internal Server Error"
+var authorizationError string = "Not Authorized"
```
Comment (masonc08) [inline api/status/response.go]:
```suggestion
const validationError string = "Validation Error"
const serverError string = "Internal Server Error"
const authorizationError string = "Not Authorized"
```

File: api/status/response.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		json.NewEncoder(w).Encode(&errorMap)
+	case 500:
+		if !ok {
+			errorMap["Error"] = serverError
+		}
+		json.NewEncoder(w).Encode(&errorMap)
+	default:
+		errorMap["Error"] = serverError
+		json.NewEncoder(w).Encode(&errorMap)
+	}
```
Comment (masonc08) [inline api/status/response.go:39]: I think the cases should be indented in by 2 spaces

File: api/endpoints/login.go [hunk 3]
```diff
@@ -15,22 +16,19 @@ func Login(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	user := new(models.User) // request user
 	if err := json.NewDecoder(r.Body).Decode(&user); err != nil {
-		w.WriteHeader(400)
-		json.NewEncoder(w).Encode(&map[string]string{"Error": err.Error()})
+		status.RespondStatus(w, 400, err)
 		return
 	}
 	var User models.User // user in database
```
Comment (masonc08) [inline api/endpoints/login.go]: Is there a specific reason why this is named `User`? It's capitalized and is similar to `user`, doesn't seem ideal to me. Would be better off renaming this to something easier to read, unless there's a special reason why we named it `User` of course

**Review by masonc08 — CHANGES_REQUESTED** [review]
Reviewed some for now, will do more research on the JWT stuff later so I understand it better

File: api/endpoints/refresh.go [hunk 4]
```diff
@@ -3,24 +3,33 @@ package endpoints
...
 
 	"github.com/drew138/tictac/api/authorization"
+	"github.com/drew138/tictac/api/status"
 )
 
 // RefreshJWT - function handle to refresh jwts
 func RefreshJWT(w http.ResponseWriter, r *http.Request) {
-	rToken := strings.Split(r.Header.Get("Authorization"), " ")[1]
-	parsedRToken, err := authorization.ParseJWT(rToken)
+	w.Header().Set("Content-Type", "application/json")
```
Comment (masonc08) [inline api/endpoints/refresh.go:13]: Do we need this line?

File: api/endpoints/changePass.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+	if User.ID != claims["ID"] {
+		status.RespondStatus(w, 500, err)
+		return
+	}
+	tokenPair, err := authorization.GenerateJWTS(&User)
+	if err != nil {
+		status.RespondStatus(w, 500, err)
+		return
+	}
+	w.WriteHeader(201)
```
Comment (masonc08) [inline api/endpoints/changePass.go]:
```suggestion
	w.WriteHeader(200)
```

File: api/status/response.go (hunk 2 above)
Comment (masonc08) [reply api/status/response.go:39]: Skip

File: api/endpoints/login.go (hunk 3 above)
Comment (masonc08) [reply api/endpoints/login.go]: Fixed already

File: api/routes.go [hunk 6]
```diff
@@ -17,7 +17,7 @@ func RegisterRoutes(r *mux.Router) {
 	// PUT Endpoints
 
 	// PATCH Endpoints
-
+	r.HandleFunc("/api/v1/changepass/{newPassword:}", endpoints.ChangePassword).Methods("PATCH")
```
Comment (masonc08) [inline api/routes.go]: Consider changing into something more secure in future

**Review by masonc08 — APPROVED** [review]
lgtm, some issues I commented on but im good to merge for now

File: api/routes.go (hunk 6 above)
Comment (drew138) [reply api/routes.go]: removed parameter in recent commit. New password will be extracted from json body

**Pull request merged**